plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1129,9 +1132,18 @@ public void alterTransactionalTable(Table table, long transactionId, long writeI
                     .stopOn(InvalidOperationException.class, MetaException.class)
                     .stopOnIllegalExceptions()
                     .run("alterTransactionalTable", stats.getAlterTransactionalTable().wrap(() -> {
-                        try (ThriftMetastoreClient client = createMetastoreClient()) {
-                            client.alterTransactionalTable(table, transactionId, writeId, new EnvironmentContext());
-                        }
+                        alternativeCall(
+                                () -> createMetastoreClient(),
+                                exception -> !isUnknownMethodExceptionalResponse(exception),
+                                chosenAlterTableAlternative,
+                                client -> {
+                                    client.alterTransactionalTable(table, transactionId, writeId, new EnvironmentContext());
+                                    return null;
+                                },
+                                client -> {
+                                    client.alterTableWithEnvironmentContext(table.getDbName(), table.getTableName(), table, new EnvironmentContext());
```

Comment:
why no `transactionId` and `writeId` being passed here here?

this is called from `updateTableWriteId` method, so seems like `writeId` is _the reason_ why we are calling HMS.

cc @djsstarburst @losipiuk

Reply:
I think the reason that there is no writeId passed explicitly here is that there is a writeId in the `Table` object.  I verified that is true by looking at the generated Thrift code.